### PR TITLE
Update RobustFluctuationObservables.cxx

### PR DESCRIPTION
**Pull request by altsybee**
added Occupancy check of K0S vs centrality

### CODE REVIEW

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 1]
```diff
@@ -63,6 +63,21 @@ using namespace o2::framework::expressions;
   mHist2D[hName] = v2D.size();                        \
   v2D.push_back(histosEventNew.add<TH2>(TString::Format("%s/%s", hName, cutName.c_str()).Data(), title, kTH2F, {axisX, axisY}));
 
+// pt study:
+#define ADD_PT_HIST_1D(hName, cutName, hTitle, axisX...)                                                                         \
+  {                                                                                                                              \
+    string str = hName;                                                                                                          \
+    string fullName = str + "/" + cutName;                                                                                       \
+    cout << "fullName = " << fullName << endl;                                                                                   \
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:71]: Please, for the next iteration remove `cout` and use `LOGF` instead if needed

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 2]
```diff
@@ -164,6 +182,13 @@ struct RobustFluctuationObservables {
   vector<vector<std::shared_ptr<TH1>>> pHist1D;
   vector<vector<std::shared_ptr<TH2>>> pHist2D;
 
+  // QA for pt vs various cuts
+  map<string, int> mPtStudyCuts;
+  // map<string, int> mTrackType;
+  // vector<vector<std::shared_ptr<TH1>>> pHistPtStudy1D;
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:188]: Please, remove commented code in the next iteration

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 3]
```diff
@@ -586,16 +681,128 @@ struct RobustFluctuationObservables {
...
+    // histosDetailedPt.add("pt_eta08_ITS7cls_noROFborder", "pt_eta08_ITS7cls_noROFborder", kTH1D, {axisLogPt});
+
+    // // .. weighted with nTPCcls
+    // histosDetailedPt.add("pt_eta08_All_ROFborder_w_nTPCcls", "pt_eta08_All_ROFborder_w_nTPCcls", kTH1D, {axisLogPt});
+    // histosDetailedPt.add("pt_eta08_Global_ROFborder_w_nTPCcls", "pt_eta08_Global_ROFborder_w_nTPCcls", kTH1D, {axisLogPt});
+    // histosDetailedPt.add("pt_eta08_ITS7cls_ROFborder_w_nTPCcls", "pt_eta08_ITS7cls_ROFborder_w_nTPCcls", kTH1D, {axisLogPt});
+
+    // histosDetailedPt.add("pt_eta08_All_noROFborder_w_nTPCcls", "pt_eta08_All_noROFborder_w_nTPCcls", kTH1D, {axisLogPt});
+    // histosDetailedPt.add("pt_eta08_Global_noROFborder_w_nTPCcls", "pt_eta08_Global_noROFborder_w_nTPCcls", kTH1D, {axisLogPt});
+    // histosDetailedPt.add("pt_eta08_ITS7cls_noROFborder_w_nTPCcls", "pt_eta08_ITS7cls_noROFborder_w_nTPCcls", kTH1D, {axisLogPt});
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:756]: Same comment here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 4]
```diff
@@ -586,16 +681,128 @@ struct RobustFluctuationObservables {
...
+        ADD_PT_HIST_1D("ITS567", cutName, "", axisLogPt);
+        ADD_PT_HIST_1D("ITS567_TPC80cl", cutName, "", axisLogPt);
+
+        ADD_PT_HIST_1D("ITS67", cutName, "", axisLogPt);
+        ADD_PT_HIST_1D("ITS67_TPC80cl", cutName, "", axisLogPt);
+
+        // add vector with pointers to histos for a given cut to an "external" vector
+        // pHistPtStudy1D.push_back(v1D);
+      }
+    }
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:789]: ... and in the corresponding parts of here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 5]
```diff
@@ -1216,12 +1523,14 @@ struct RobustFluctuationObservables {
     }
 
     // ### event selection cuts
-    if (!collision.sel8()) {
+    // if (!collision.sel8()) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1526]: Please, don't leave code commented

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 6]
```diff
@@ -1216,12 +1523,14 @@ struct RobustFluctuationObservables {
...
+    if (!collision.selection_bit(o2::aod::evsel::kIsTriggerTVX)) {
       prevBC = collBC;
       prevBcInTF = bcInTF;
       prevOrbit = orbitInTF;
       prevTF = TFid;
       prevGlobalFoundBC = globalFoundBC;
+      prevFoundBcInTF = foundBcInTF;
       // double prev_vZ_keepForBelow = prev_vZ;
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1534]: Same comment here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 7]
```diff
@@ -1287,16 +1604,17 @@ struct RobustFluctuationObservables {
       int64_t nextBcInTF = (bcNext.globalBC() - bcSOR) % nBCsPerTF;
 
       // found BC
-      uint64_t globalNextFoundBC = 9999; //-1;
+      // uint64_t globalNextFoundBC = 9999; //-1;
       // if (nextColl.has_foundFT0() && nextColl.has_foundBC() && collision.selection_bit(o2::aod::evsel::kNoTimeFrameBorder) && nextColl.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1608]: And here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 8]
```diff
@@ -1287,16 +1604,17 @@ struct RobustFluctuationObservables {
...
       if (nextColl.selection_bit(o2::aod::evsel::kIsTriggerTVX) && nextColl.has_foundBC() && collision.selection_bit(o2::aod::evsel::kNoTimeFrameBorder) && nextColl.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
-        auto nextBcFound = nextColl.foundBC_as<BCsRun3>(); // collision.foundBC();
-        globalNextFoundBC = nextBcFound.globalBC() % 3564;
+        // auto nextBcFound = nextColl.foundBC_as<BCsRun3>(); // collision.foundBC();
+        // globalNextFoundBC = nextBcFound.globalBC() % 3564;
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1611]: And here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 9]
```diff
@@ -1287,16 +1604,17 @@ struct RobustFluctuationObservables {
...
       }
 
       // check if Found BC is NOT the same for prev, current and next events:
       // if (globalFoundBC >= 0 && globalNextFoundBC >= 0 && prevGlobalFoundBC >= 0 && globalFoundBC != globalNextFoundBC && globalFoundBC != prevGlobalFoundBC) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1615]: and here

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 10]
```diff
@@ -1287,16 +1604,17 @@ struct RobustFluctuationObservables {
...
-      if (globalFoundBC != 9999 && globalNextFoundBC != 9999 && prevGlobalFoundBC != 9999 && globalFoundBC != globalNextFoundBC && globalFoundBC != prevGlobalFoundBC) {
+      // if (globalFoundBC != 9999 && globalNextFoundBC != 9999 && prevGlobalFoundBC != 9999 && globalFoundBC != globalNextFoundBC && globalFoundBC != prevGlobalFoundBC) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1616]: ...

File: PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx [hunk 11]
```diff
@@ -1555,7 +1877,8 @@ struct RobustFluctuationObservables {
     // fillHistForThisCut("hasFT0_CorrectedValid", multNTracksPV, multTrk, nTracksGlobalAccepted, multT0A, multT0C, multV0A, t0cCentr, collBC);
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx:1877]: ...

**Review by victor-gonzalez — APPROVED** [review]
I approve it for the time being but, please, have a look at my comments for the next iteration

**Comment by altsybee** [comment]
> I approve it for the time being but, please, have a look at my comments for the next iteration

Hello Victor, thanks a lot, I will correct the code accordingly

**Pull request merged**